### ISSUE TITLE
value or text on the definition of an SVG selector

### ISSUE THREAD

**Issue by iherman** [w3c/web-annotation#191]
(This issue is _not_ aimed at the 2016-03-31 publication, it can be handled for future releases!)

At present, the SVGSelector has a the following structure:

Either the svg is referred to via a URI; in JSON this means that it uses the id:

``` json
"selector" : {
   "type" : "SvgSelector",
   "id" : "http://ex.org/svg"
}
```

or the svg part in incorporated in the object, ie,

``` json
"selector" : {
   "type": ["Selector","Content"],
   "text": "<svg:svg> ... </svg:svg>"
}
```

I have several problems with this
1. Modeling-wise, in terms of RDF, I believe what we have here is that we add an `rdf:type` to the `http://ex.org/svg`. I do not think this is the right thing to do, it looks a little bit like type hijacking (we may not "own" the `http://ex.org/svg`, adding an extra type is not really our thing to do)
2. This is the only place in selectors where an extra type should be used.
3. Everywhere else we use "value" when we specify the details of a Selector, the usage of 'text' is a bit odd.

What about:

``` json
"selector" : {
   "type" : "SvgSelector",
   "source" : "http://ex.org/svg"
}
```

and

``` json
"selector" : {
   "type" : "SvgSelector",
   "value": "<svg:svg> ... </svg:svg>"
}
```

This seems to be more consistent with the rest

**Comment by azaroth42** [comment]
It's consistent with the embedded content pattern, rather than the values in selectors pattern.  The use of text here is because we used to use Content in RDF to embed the SVG resource's representation. The same with the CSS Style.  I think this is correct, rather than value ... we're embedding the content of the representation into the graph, rather than providing a stand alone value that is not a representation.

If there isn't an id, then it's just a blank node.  So long as the representation _is_ available from the URI in id, I don't see a problem. 

That said I do see the point about type-jacking. We solved that already for Semantic Tags where the resources were very likely not to be ours.  On the other hand, here it seems that finding an SVG document in the wild that happened to express the right area for the selection is so improbable as to not be worth worrying about?

**Comment by iherman** [comment]
> It's consistent with the embedded content pattern, rather than the values in selectors pattern. The use of text here is because we used to use Content in RDF to embed the SVG resource's representation.
> 
> Well, that is past. Not relevant any more for this version...
> The same with the CSS Style. I think this is correct, rather than value ... we're embedding the content of the representation into the graph, rather than providing a stand alone value that is not a representation.

I do not find this argument convincing. For the selector, the _role_ of the textual svg content is the same as, say, the xpath content for the XPath Selector: it is the value that governs the selection for an XPath selection. The fact that we are talking about some content that, _in other contexts_ is interpretable as a stand alone image, is irrelevant.

Note that the "Content" class ('pattern', if you refer to) appears twice. One is the SVG Selector, the other is Style (I admit I did not look at the latter for my comment). But for Style we use `value` and not `text`, so we _are_ inconsistent. B.t.w., in the Style element there is also a `format` argument, that is irrelevant or, more exactly, superfluous, because we are talking about a CssStylesheet object!

I have the impression that the Content class itself should be therefore removed from the model altogether, and simplify both usages. It is an overcomplication without any benefit that I would see.

> If there isn't an id, then it's just a blank node. So long as the representation is available from the URI in id, I don't see a problem.
> 
> I am not sure what you refer to.
> 
> That said I do see the point about type-jacking. We solved that already for Semantic Tags where the resources were very likely not to be ours. On the other hand, here it seems that finding an SVG document in the wild that happened to express the right area for the selection is so improbable as to not be worth worrying about?
> 
> It is again a matter of consistency. The Open Web Assumption has already directed our deliberation in the past, and we decided to be consistent with it (see our discussion on roles). We should apply the same here.

Bottom line: I stand by my original proposal...

**Comment by azaroth42** [comment]
We should definitely be consistent between SvgSelector and CssStyle, that's definitely a bug regardless of the outcome of this discussion.

It seems the same as a textual body to me as well, from a protocol perspective.  I can either retrieve the body/stylesheet/svg by dereferencing the IRI supplied in `id`, or I can use the content provided in the graph.  If all three are consistent, then I'm happy... so ...

To extend your proposal, can we drop everything down to using the more generic `value` and get rid of `text`, `TextualBody` and `Content` completely?

e.g. as above, plus change this:

```
{
  "type": "Annotation",
  "body": {
    "type": "TextualBody",    // oa:TextualBody
    "format": "text/plain",
    "text": "I love this page!"   // oa:text
  },
  "target": "http://example.com/index.html"
}
```

to:

```
{
 "type": "Annotation",
 "body": {
    "type": "Text",    //dcmitypes:Text
    "format": "text/plain",
    "value": "I love this page!"    // rdf:value
  },
  "target": "http://example.com/index.html"
}
```

**Comment by iherman** [comment]
> On 28 Mar 2016, at 18:23, Rob Sanderson notifications@github.com wrote:
> 
> We should definitely be consistent between SvgSelector and CssStyle, that's definitely a bug regardless of the outcome of this discussion.
> 
> It seems the same as a textual body to me as well, from a protocol perspective. I can either retrieve the body/stylesheet/svg by dereferencing the IRI supplied in id, or I can use the content provided in the graph. If all three are consistent, then I'm happy... so ...
> 
> To extend your proposal, can we drop everything down to using the more generic value and get rid of text, TextualBody and Content completely?
> 
> That works for me. One additional thing: 'format' might be optional. It is probably fine for the Text, but for, say, an SVGSelector or a CssStyle it is obvious that the value is SVG, resp. CSS, ie, adding a format at that point seems to be unnecessary

Ivan

> e.g. as above, plus change this:
> 
> {
>   "type": "Annotation",
>   "body": {
>     "type": "TextualBody",    // oa:TextualBody
>     "format": "text/plain",
>     "text": "I love this page!"   // oa:text
>   },
>   "target": "http://example.com/index.html"
> }
> to:
> 
> {
>  "type": "Annotation",
>  "body": {
>     "type": "Text",    //dcmitypes:Text
>     "format": "text/plain",
>     "value": "I love this page!"    // rdf:value
>   },
>   "target": "http://example.com/index.html"
> }
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub https://github.com/w3c/web-annotation/issues/191#issuecomment-202470464

---

Ivan Herman, W3C
Digital Publishing Lead
Home: http://www.w3.org/People/Ivan/
mobile: +31-641044153
ORCID ID: http://orcid.org/0000-0003-0782-2704

**Comment by azaroth42** [comment]
`format` should definitely be optional for SvgSelector and CssStyle :)

**Comment by azaroth42** [comment]
Would we also be okay with the proposal for situations when there /is/ a URI?

```
{
  "type": "Annotation",
  "target": "http://example.com/index.html",
  "body": {
    "id": "http://example.org/texts/t1",
    "type": "Text",
    "value": "The text representation obtained by dereferencing id",
    "format": "text/plain"
  }
}
```

**Comment by iherman** [comment]
@azaroth42 :

> Would we also be okay with the proposal for situations when there /is/ a URI?
> 
> ```
> {
>  "type": "Annotation",
>  "target": "http://example.com/index.html",
>  "body": {
>    "id": "http://example.org/texts/t1",
>    "type": "Text",
>    "value": "The text representation obtained by dereferencing id",
>    "format": "text/plain"
>  }
> }
> ```

I do not see why not, although I am not sure what the practical usage of it would be. (From a model point of view, the subject for "body" is, conceptually, an anonymous node, ie, having a blank node seems to be natural.)

**Comment by iherman** [comment]
Discussed on telco 2016.05.06, resolved using `rdf:value` everywhere

See: http://www.w3.org/2016/05/06-annotation-irc#T15-49-12-1

**Comment by iherman** [comment]
Also addressed: SvgSelector does not need the extra "Content" type and format is not necessary either (also relevant for the CssSelector)

See http://www.w3.org/2016/05/06-annotation-irc#T15-53-17 and http://www.w3.org/2016/05/06-annotation-irc#T15-53-47
